.github/workflows/test-build-push.yml

```diff
@@ -0,0 +1,106 @@
+name: Run tests, build image and push to GitHub Container Registry
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  golangci-lint:
+    name: Run golangci-lint
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version: "1.22.3"
+
+      - name: Setup Git Authentication
+        run: |
+          git config --global url."https://${{ github.token }}:x-oauth-basic@github.com/".insteadOf "https://github.com/"
+
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: latest
+          args: --timeout 10m --verbose
+
+  gocritic:
+    name: Run gocritic
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version: "1.22.3"
+
+      - name: Run gocritic
+        run: |
+          go install github.com/go-critic/go-critic/cmd/gocritic@v0.11.2
+          gocritic check -disable=sloppyTypeAssert ./...
+
+  test:
+    name: Run unit tests
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version: "1.22.3"
+
+      - name: Setup Git Authentication
+        run: |
+          git config --global url."https://${{ github.token }}:x-oauth-basic@github.com/".insteadOf "https://github.com/"
+
+      - name: Run unit tests
+        run: CGO_ENABLED=0 go test ./... -short
+
```

Comment:
Any reason we can't just run the E2E test job in this workflow and remove the separate file?

Reply:
OK fine with me. I generally prefer waiting a bit longer from more confidence from more comprehensive tests but I'm fine with keeping them separate. I can always just run the E2E locally.

Reply:
I've arranged the jobs in a way where unit testing and e2e run in parallel, but the build job requires both of them to be successful before running the build. That would give us confidence that both types of tests get executed, taking less CI time and ensuring that any of them failing will not produce a non-functional image.